text-generation-inference/server/text_generation_server/generator.py

```diff
@@ -429,6 +429,12 @@ def decode(self, batches: List[CachedBatch]) -> Tuple[List[Generation], CachedBa
         for i, slot in enumerate(self.slots):
             if slot.state != Slot.State.EMPTY:
                 if input_ids is None:
+                    pad_token_id = self.tokenizer.pad_token_id
+                    if pad_token_id is None:
+                        if isinstance(self.tokenizer.eos_token_id, list):
+                            pad_token_id = self.tokenizer.eos_token_id[0]
+                        else:
+                            pad_token_id = self.tokenizer.eos_token_id
```

Comment:
This block doesn't seem to depend on `i` should we move it outside the for loop?